R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+    #ISO codes of country file
+    iso<-substr(tools::file_path_sans_ext(basename(files[f])), 0, 3)
+
+
+    #if ISO codes is not in initilsed list of interest skip this file, this could be defined already in template dictionary ?
+    if(!(iso %in% isotomerge)){
+      print(paste(iso,"skip"))
+      next
+      }
+
+    print(iso)
+
+    #for this iso code set parameters dictating file load in and clean process using the template_check dictionary
+    sheetname<-template_check$sheetname[template_check$country==iso]
+    skip<-template_check$skip[template_check$country==iso]
+    template<-template_check$template[template_check$country==iso]
+
+
+    # load in file
+    #files that are xlsb (CIV) cannot be open (easily on mac) for ease resave the file in an xlsx format, this can be edited to use the RODBC package which i think works on windows
+    if (grepl(files[f],pattern="\\.xlsb$", ignore.case=TRUE)){
+      print(paste(iso, "file is in xlsb format, please resave file in xlsx format before proceeding"))
+    }else{
+      # og_sheet<- openxlsx::read.xlsx(files[f],sheet=sheetname, startRow=skip+1,detectDates=T, skipEmptyRows = F,
+      #                                skipEmptyCols = F)
+      og_sheet <- rio::import(files[f], which=sheetname, skip=skip)
+    }
+
+    # clean variable names to removed unwated regex patterns, replace all spaces with a ., inorder to match var_dict dictionary
+    names(og_sheet) <- gsub(x= names(og_sheet),pattern ="\r\n",replacement = "", fixed = T)
+    names(og_sheet) <-gsub(x= names(og_sheet),pattern ="\\s+", replacement =" ")
+    names(og_sheet) <-gsub(x= names(og_sheet),pattern ="\\s+$", replacement ="")
+    names(og_sheet) <-gsub(x= names(og_sheet),pattern ="\\s+|[[:punct:]]", replacement ="_")
+    names(og_sheet) <-tolower(names(og_sheet))
+    names(og_sheet)<- stringi::stri_trans_general(names(og_sheet) , "Latin-ASCII")
+
+    #remove all accents from values in dataframe
+    #replace all values in file that should be coded as missing
+    #using gsub and regex pattern matching here as some patterns require to be fixed and others not
+    og_sheet<-og_sheet %>% mutate_all(stringi::stri_trans_general,"Latin-ASCII") %>%
+      mutate_all(gsub,pattern="['?]",replacement = "",ignore.case = T, perl = T) %>%
+      mutate_all(gsub, pattern="(?i)^NA$|(?i)^N/A$|(?i)^N/A,|(?i)^N\\A$|(?i)^Unknown$|(?i)^dont know$|(?i)^Unkown$|(?i)^N.A$|(?i)^NE SAIT PAS$|(?i)^inconnu$|^ $|(?i)^Nao aplicavel$|(?i)^Sem informacao$", replacement=NA, perl = T)
+#must keep this pattern all one line or doesnt work
+
+    # filter variable cleaning dictionary specific to country file loaded
+    var_dict_country<- var_dict %>% filter(country==iso)
+    var_dict_country<- var_dict_country[var_dict_country$old_var_r %in% names(og_sheet),]
+
+    # list variables to drop that are non-template
+    drop_nontemplate_vars<- as.list(var_dict_country$old_var_r[var_dict_country$notes=="non_template" & !is.na(var_dict_country$notes)])
```

Comment:
dont think this needs to be as.list can just be a vector and then use !which instead of -which